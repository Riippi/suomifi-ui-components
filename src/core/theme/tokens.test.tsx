import React from 'react';
import { render } from 'react-testing-library';
import { axeTest } from '../../utils/test/axe';

import { Panel, defaultTokens } from '../../';

const { colors } = defaultTokens;
const customColors = {
  ...colors,
  highlightBase: { hsl: 'hsl(1, 2%, 3%)', h: 1, s: 2, l: 3 },
};

const Test = (
  <Panel.expansionGroup
    OpenAll={'Open all'}
    CloseAll={'Close all'}
    tokens={{ colors: customColors }}
  >
    <Panel.expansion title="Test expansion 1">
      Test expansion content 1
    </Panel.expansion>
    <Panel.expansion title="Test expansion 2">
      Test expansion content 2
    </Panel.expansion>
    <Panel.expansion title="Test expansion 3">
      Test expansion content 3
    </Panel.expansion>
  </Panel.expansionGroup>
);

test('snapshot testing', () => {
  const TestRendered = render(Test);
  const { container } = TestRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test('should not have basic accessibility issues', axeTest(Test));
