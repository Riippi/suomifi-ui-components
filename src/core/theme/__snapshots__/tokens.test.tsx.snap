// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c2:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c2::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c2::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c2::-webkit-inner-spin-button {
  height: auto;
}

.c2::-webkit-outer-spin-button {
  height: auto;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c3 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c7 {
  display: inline-block;
  vertical-align: baseline;
}

.c4 {
  position: absolute;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
}

.c1 {
  color: hsl(0,0%,16%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  max-width: 100%;
}

.c1 > .fi-expander-group_expanders {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c1 > .fi-expander-group_expanders .fi-expander {
  margin-top: 0;
  margin-bottom: 0;
  border-radius: 0;
  border-top: 1px solid hsl(202,7%,80%);
  -webkit-transition: margin 50ms cubic-bezier(0.28,0.84,0.42,1);
  transition: margin 50ms cubic-bezier(0.28,0.84,0.42,1);
}

.c1 > .fi-expander-group_expanders .fi-expander .fi-icon {
  color: hsl(1,2%,3%);
}

.c1 > .fi-expander-group_expanders .fi-expander:first-child {
  border-radius: 2px 2px 0 0;
  border-top: none;
}

.c1 > .fi-expander-group_expanders .fi-expander:last-child {
  border-radius: 0 0 2px 2px;
}

.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open {
  border-top: none;
}

.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open:not(:first-of-type) {
  margin-top: 16px;
}

.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open:not(:last-of-type) {
  margin-bottom: 16px;
}

.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open:first-child,
.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open:first-child:before {
  border-radius: 2px 2px 0 0;
}

.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open:last-child,
.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open:last-child:before {
  border-radius: 0 0 2px 2px;
}

.c1 > .fi-expander-group_expanders .fi-expander.fi-expander--open + .fi-expander {
  border-top: none;
}

.c1 > .fi-expander-group_all-button {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-left: auto;
  margin-bottom: 8px;
  padding: 4px 0;
  color: hsl(1,2%,3%);
  border-radius: 2px;
  border: none;
  cursor: pointer;
}

.c1 > .fi-expander-group_all-button:focus {
  outline: 0;
  position: relative;
}

.c1 > .fi-expander-group_all-button:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c5 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  background-color: hsl(0,0%,100%);
  position: relative;
  padding: 0;
  border-radius: 2px;
  box-shadow: 0 1px 2px 0 rgba(41,41,41,0.14),0 1px 5px 0 rgba(41,41,41,0.12);
  display: block;
  width: 100%;
  max-width: 100%;
}

.c5:before {
  background-color: hsl(212,63%,98%);
  opacity: 0;
}

.c8 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  background-color: hsl(0,0%,100%);
  position: relative;
  visibility: hidden;
  display: block;
  height: 0;
  overflow: hidden;
  word-break: break-word;
  -webkit-transform: scaleY(0);
  -ms-transform: scaleY(0);
  transform: scaleY(0);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-transition: all 50ms cubic-bezier(0.28,0.84,0.42,1);
  transition: all 50ms cubic-bezier(0.28,0.84,0.42,1);
  will-change: transition,height;
}

.c8:not(.fi-expander_content--no-padding) {
  padding: 0 16px;
}

.c8.fi-expander_content--open {
  visibility: visible;
  height: 10%;
  overflow: visible;
  -webkit-animation: fi-expander_content-anim 50ms cubic-bezier(0.28,0.84,0.42,1) 1 forwards;
  animation: fi-expander_content-anim 50ms cubic-bezier(0.28,0.84,0.42,1) 1 forwards;
}

.c8.fi-expander_content--open:not(.fi-expander_content--no-padding) {
  padding-top: 0;
  padding-right: 20px;
  padding-bottom: 20px;
  padding-left: 20px;
}

.c6 {
  color: hsl(0,0%,16%);
  position: relative;
  display: block;
  width: 100%;
  max-width: 100%;
  min-height: 60px;
}

.c6 .fi-expander_title-button {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  background-color: hsl(0,0%,100%);
  color: hsl(212,63%,45%);
  display: block;
  font-size: font-family:'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
  width: 100%;
  max-width: 100%;
  min-height: 60px;
  padding: 17px 60px 16px 20px;
}

.c6 .fi-expander_title-button:focus {
  outline: 0;
}

.c6 .fi-expander_title-button:focus-within {
  outline: 0;
}

.c6 .fi-expander_title-button:focus-within:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c6 .fi-expander_title-button,
.c6 .fi-expander_title-button * {
  cursor: pointer;
}

.c6 .fi-expander_title-button:hover,
.c6 .fi-expander_title-button:active,
.c6 .fi-expander_title-button:focus,
.c6 .fi-expander_title-button:focus-within {
  cursor: pointer;
}

.c6 .fi-expander_title-icon {
  position: absolute;
  height: 20px;
  width: 20px;
  top: 0;
  right: 0;
  margin: 20px;
}

.c6 .fi-expander_title--open .fi-expander_title-icon,
.c6 .fi-expander_title-icon--open {
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

<div
  class="c0 c1 fi-expander-group"
>
  <button
    class="c2 fi-expander-group_all-button"
    type="button"
  >
    <span
      aria-hidden="true"
      class="c3"
    >
      Open all
    </span>
    <span
      class="c3 c4 fi-visually-hidden"
    >
      Open all
    </span>
  </button>
  <div
    class="c0 fi-expander-group_expanders"
  >
    <div
      class="c0 c5 fi-expander"
      id="id-expander-1"
    >
      <div
        class="c0 c6 fi-expander_title"
      >
        <button
          aria-controls="id-expander-1_content"
          aria-expanded="false"
          class="c2 fi-expander_title-button"
          id="id-expander-1_title"
          type="button"
        >
          Test expander 1
          <svg
            aria-hidden="true"
            class="fi-icon c7 fi-expander_title-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M13 15.6c-.3.3-.6.4-1 .4s-.7-.1-1-.4l-5.6-5.3c-.5-.5-.5-1.4 0-1.9s1.4-.5 2 0l4.6 4.4 4.6-4.4c.5-.5 1.4-.5 2 0 .5.5.5 1.4 0 1.9L13 15.6z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c0 c5 fi-expander"
      id="id-expander-2"
    >
      <div
        class="c0 c6 fi-expander_title"
      >
        <button
          aria-controls="id-expander-2_content"
          aria-expanded="false"
          class="c2 fi-expander_title-button"
          id="id-expander-2_title"
          type="button"
        >
          Test expander 2
          <svg
            aria-hidden="true"
            class="fi-icon c7 fi-expander_title-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M13 15.6c-.3.3-.6.4-1 .4s-.7-.1-1-.4l-5.6-5.3c-.5-.5-.5-1.4 0-1.9s1.4-.5 2 0l4.6 4.4 4.6-4.4c.5-.5 1.4-.5 2 0 .5.5.5 1.4 0 1.9L13 15.6z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="id-expander-2_title"
        class="c0 c8 fi-expander_content"
        id="id-expander-2_content"
      >
        Test expander content 2
      </div>
    </div>
    <div
      class="c0 c5 fi-expander"
      id="id-expander-3"
    >
      <div
        class="c0 c6 fi-expander_title"
      >
        <button
          aria-controls="id-expander-3_content"
          aria-expanded="false"
          class="c2 fi-expander_title-button"
          id="id-expander-3_title"
          type="button"
        >
          Test expander 3
          <svg
            aria-hidden="true"
            class="fi-icon c7 fi-expander_title-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M13 15.6c-.3.3-.6.4-1 .4s-.7-.1-1-.4l-5.6-5.3c-.5-.5-.5-1.4 0-1.9s1.4-.5 2 0l4.6 4.4 4.6-4.4c.5-.5 1.4-.5 2 0 .5.5.5 1.4 0 1.9L13 15.6z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="id-expander-3_title"
        class="c0 c8 fi-expander_content"
        id="id-expander-3_content"
      >
        Test expander content 3
      </div>
    </div>
  </div>
</div>
`;
